examples/typescript/vercel-ai-chatbot-ollama/lib/chat/actions.tsx

```diff
@@ -0,0 +1,371 @@
+import 'server-only'
+
+// Import necessary functions and components
+import {
+  createAI,
+  createStreamableUI,
+  getMutableAIState,
+  getAIState,
+  streamUI,
+  createStreamableValue
+} from 'ai/rsc'
+import { ollama } from 'ollama-ai-provider'
+
+import {
+  spinner,
+  BotCard,
+  BotMessage,
+  SystemMessage,
+  Stock,
+  Purchase
+} from '@/components/stocks'
+
+import { z } from 'zod'
+import { EventsSkeleton } from '@/components/stocks/events-skeleton'
+import { Events } from '@/components/stocks/events'
+import { StocksSkeleton } from '@/components/stocks/stocks-skeleton'
+import { Stocks } from '@/components/stocks/stocks'
+import { StockSkeleton } from '@/components/stocks/stock-skeleton'
+import {
+  formatNumber,
+  runAsyncFnWithoutBlocking,
+  sleep,
+  nanoid
+} from '@/lib/utils'
+import { saveChat } from '@/app/actions'
+import { SpinnerMessage, UserMessage } from '@/components/stocks/message'
+import { Chat, Message } from '@/lib/types'
+import { auth } from '@/auth'
+import { MemoizedReactMarkdown } from '@/components/markdown'
+
+// Function to submit user messages
+async function submitUserMessage(content: string) {
+  'use server'
+
+  const aiState = getMutableAIState<typeof AI>()
+
+  // Update AI state with the new user message
+  aiState.update({
+    ...aiState.get(),
+    messages: [
+      ...aiState.get().messages,
+      {
+        id: nanoid(),
+        role: 'user',
+        content
+      }
+    ]
+  })
+
+  let textStream: undefined | ReturnType<typeof createStreamableValue<string>>
+  let textNode: undefined | React.ReactNode
+
+  // Stream UI for the AI response
+  const result = await streamUI({
+    model: ollama('mannix/llama3.1-8b-lexi:tools-q8_0'), // Use Ollama model
```

Comment:
@mhioi 
cool!

any chance you could let the user specify the model in the UI? somehow

Reply:
I think that could be done, I'll try my best.

I was thinking about adding a feature to let user see his screen records or screen frames in UI.

Reply:
just do a free from input and below write "only works with models that supports tools"

Reply:
ok, I will leave a free text input to let user specify the model.